modules/mauth-common/src/test/scala/com/mdsol/mauth/MAuthSignatureHelperSpec.scala

```diff
@@ -89,8 +89,8 @@ class MAuthSignatureHelperSpec extends AnyFlatSpec with Matchers {
   }
 
   it should "correctly encode query string with special chars" in {
-    val queryString = "key=-_.~ !@#$%^*()+{}|:\"'`<>?"
-    val expectedString = "key=-_.~%20%21%40%23%24%25%5E%2A%28%29%2B%7B%7D%7C%3A%22%27%60%3C%3E%3F"
+    val queryString = "key=-_.~!@#$%^*()+{}|:\"'`<>?"
+    val expectedString = "key=-_.~%21%40%23%24%25%5E%2A%28%29%20%7B%7D%7C%3A%22%27%60%3C%3E%3F"
```

Comment:
This test seems wrong. Its input should already be encoded query param but `queryString` definitely isn't encoded.

An example expected input: `key2=asdf+f&key=%21%40%23` and the expected output `key=%21%40%23&key2=asdf%20` (`+` is decoded as space, and then reencoded into `%20` according to the spec)

Reply:
updated